include/roboteam_ai/control/controllers/PidController.h

```diff
@@ -0,0 +1,144 @@
+//
+// Created by mrlukasbos on 29-3-19.
+//
+
+#ifndef ROBOTEAM_AI_PID_H
+#define ROBOTEAM_AI_PID_H
+
+#include <tuple>
+#include "BaseController.h"
+
+/**
+ * The basic implementation of a PIDF (proportional - integrative - derivative
+ * - feedback) controller. The actual controller accepts a float reference
+ * and outputs a float controller action. For more info check https://en.wikipedia.org/wiki/PID_controller
+ */
+class PidController : BaseController<double,double>{
+public:
+    /** Create a new PID object.
+     * @param p Proportional gain. Large if large difference between setpoint and target.
+     * @param i Integral gain.	Becomes large if setpoint cannot reach target quickly.
+     * @param d Derivative gain. Responds quickly to large changes in error. Small values prevents P and I terms from causing overshoot.
+     */
+    PidController(double p, double i, double d);
+    /** Create a new PID object.
+     * @param p Proportional gain. Large if large difference between setpoint and target.
+     * @param i Integral gain.	Becomes large if setpoint cannot reach target quickly.
+     * @param d Derivative gain. Responds quickly to large changes in error. Small values prevents P and I terms from causing overshoot.
+     * @param f Feedforward gain. The value that will be multiplied with the input and added to the output
+     */
+    PidController(double p, double i, double d, double f);
+    /** Create a new PID object
+     * @oaram pid tuple with the three parameters of the PID
+     */
+    explicit PidController(std::tuple<double, double, double> pid);
+    /** Create a new PID object
+     * @oaram pid tuple with the three parameters of the PID, plus feedforward gain
+     */
+    explicit PidController(std::tuple<double, double, double, double> pidf);
+
+    void setPID(double, double, double);
+    void setPID(std::tuple<double, double, double>);
+    void setPID(std::tuple<double, double, double>, double);
+
+    void setPID(double, double, double, double);
+    /**Set the maximum output value contributed by the I component of the system
+     * this->can be used to prevent large windup issues and make tuning simpler
+     * @param maximum. Units are the same as the expected output value
+     */
+    void setMaxIOutput(double maximum);
+    /**Specify a maximum output. If a single parameter is specified, the minimum is
+     * set to (-maximum).
+     * @param output
+     */
+    void setOutputLimits(double output);
+    /**
+     * Specify a maximum output.
```

Comment:
I guess you also specify minimum output, so this should be included in the summary as well